tests/baselines/reference/expandoFunctionNullishProperty.types

```diff
@@ -0,0 +1,50 @@
+//// [tests/cases/compiler/expandoFunctionNullishProperty.ts] ////
+
+=== expandoFunctionNullishProperty.ts ===
+// mentioned in https://github.com/microsoft/TypeScript/issues/54220
+
+interface TestNull {
+  (): void;
+  readonly prop: null;
+>prop : null
+}
+
+export function testNull(): TestNull {
+>testNull : () => TestNull
+
+  function inner() {}
+>inner : { (): void; prop: null; }
+
+  inner.prop = null;
+>inner.prop = null : null
+>inner.prop : null

```

Comment:
to confirm, without this change, this line would be `>inner.prop : any`, right?

It might be good to add a js file entry in this test as a comparison (although come to think of it, it's likely already tested).

Reply:
> to confirm, without this change, this line would be >inner.prop : any, right?

right

> It might be good to add a js file entry in this test as a comparison (although come to think of it, it's likely already tested).

Yeah, I think that this is already tested because at first I just removed this whole check, got baseline errors, and only then I learned about this rule being important~ for JS files. So I think that there is no need for an extra test case here but if you think otherwise - let me know.